libcore/IconInfo.vala

```diff
@@ -90,9 +90,17 @@ public class Marlin.IconInfo : GLib.Object {
 
             var names = ((GLib.ThemedIcon) icon).get_names ();
             var theme = get_icon_theme ();
-            var gtkicon_info = theme.choose_icon_for_scale (names, size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
+
+            /* Looks in default theme and hicolor theme for primary icon name first */
+            var gtkicon_info = theme.lookup_icon_for_scale (names[0], size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
```

Comment:
Why not something like this?
```suggestion
            var gtkicon_info = theme.lookup_by_gicon_for_scale (icon, size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
```

Reply:
It doesn't work, I am afraid.  It no longer finds hicolor icons in that form.  I do not know why exactly.

Reply:
@jeremypw Does PR work in this way or what else does it need to be integrated into Files?